Memory_Excercises/Excercise5/Source.cpp

```diff
@@ -0,0 +1,126 @@
+/*Exercise 5 - Dynamic Allocation for Array Sizes
+*Create a Student class that contains a char array of size 5 for a name and an int for an age. Ask
+*the user to enter how many students they want in a class. Then, initialize an array of students in
+*the heap of exactly the size they asked for. You can now probably tell how dynamic allocation is
+*useful.
+*Ask the user for a name and an age for each student and then print them all to the screen. Note
+*that you won�t be able to use names longer than 4 characters.
+*Remember to free any allocated memory.*/
+
+/**/
+
+#include <iostream>
+
+
+class FStudent
+{
+public:
+	char* NameStudent;
+	int StudentAge;
+
+	FStudent()
+	{
+		NameStudent = nullptr;
+		StudentAge = 0;
+	}
+
+	//----------------------------------------------
+	FStudent( int NewAge)
+	{
+		NameStudent = nullptr;
+		StudentAge = NewAge;
+	}
+
+	//----------------------------------------------
+	~FStudent()
+	{
+		delete NameStudent;
+	}
+
+	//----------------------------------------------
+	void SetName(const char* NewName)
+	{
+		/*Exercise 9: At this point, it�s obvious that keeping track of all memory allocations in a program can be a hassle. In the future, 
+		*we�ll discuss techniques that will make life a lot easier for us.
+		*/
+
+		if (NameStudent != nullptr)
+		{
+			NameStudent = nullptr;
+		}
+
+		int NameLenght = 0;
+		for (int i = 0; i < 32; i++)
+		{
+			if (NewName[i] != '\0')
+			{
+				NameLenght++;
+			}
```

Comment:
perhaps add a break as an else statement (?)

Reply:
Yep, better to break when there's no more need to iterate. Also, what if the name was longer than 32 characters?

The conditions in your for loop don't always need to be `i < x`. They can be whatever you want. So you can write something like 
```
for (int i = 0; NewName[i] != 0; i++)
{
    NameLength++;
}
```

That way you don't depend on some arbitrary max length
That way you don't depend on some arbitrary length